plugins/janus_sipre.c

```diff
@@ -3248,6 +3270,29 @@ static void janus_sipre_media_cleanup(janus_sipre_session *session) {
 	}
 	/* Clean up SRTP stuff, if needed */
 	janus_sipre_srtp_cleanup(session);
+
+	// media fields not cleaned up elsewhere
+	g_free(session->media.remote_audio_ip);
+	session->media.remote_audio_ip = NULL;
+	g_free(session->media.remote_video_ip);
+	session->media.remote_video_ip = NULL;
+	session->media.earlymedia = FALSE;
+	session->media.update = FALSE;
+	session->media.updated = FALSE;
+	session->media.ready = FALSE;
+	session->media.require_srtp = FALSE;
+	session->media.on_hold = FALSE;
+	session->media.has_audio = FALSE;
+	session->media.audio_pt = -1;
+	session->media.audio_pt_name = NULL;	/* Immutable string, no need to free*/
+	session->media.audio_send = TRUE;
+	session->media.pre_hold_audio_dir = JANUS_SDP_DEFAULT;
+	session->media.has_video = FALSE;
+	session->media.video_pt = -1;
+	session->media.video_pt_name = NULL;	/* Immutable string, no need to free*/
+	session->media.video_send = TRUE;
+	session->media.pre_hold_video_dir = JANUS_SDP_DEFAULT;
+	janus_rtp_switching_context_reset(&session->media.context);
```

Comment:
For all three plugins (writing it just here for the sake of brevity), it may be simpler to create a helper method that includes all that stuff, e.g., `janus_sipre_media_reset` or something like that. Doing this all inline looks ugly.